Firestore/Example/Tests/API/FIRFirestoreTests.mm

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIROptionsInternal.h>
+#import <FirebaseFirestore/FIRFirestore.h>
+
+#import <XCTest/XCTest.h>
+
+@interface FIRFirestoreTests : XCTestCase
+@end
+
+@implementation FIRFirestoreTests
+
+- (void)testDeleteApp {
+  // Create a FIRApp for testing.
+  NSString *appName = @"custom_app_name";
+  FIROptions *options =
+      [[FIROptions alloc] initWithGoogleAppID:@"1:123:ios:123ab" GCMSenderID:@"gcm_sender_id"];
+  options.projectID = @"project_id";
+  [FIRApp configureWithName:appName options:options];
+
+  // Ensure the app is set appropriately.
+  FIRApp *app = [FIRApp appNamed:appName];
+  FIRFirestore *firestore = [FIRFirestore firestoreForApp:app];
+  XCTAssertEqualObjects(firestore.app, app);
+
+  XCTestExpectation *defaultAppDeletedExpectation =
+      [self expectationWithDescription:
+                @"Deleting the default app should invalidate the default "
+                @"Firestore instance."];
+  [app deleteApp:^(BOOL success) {
+    // Recreate the FIRApp with the same name, fetch a new Firestore instance and make sure it's
+    // different than the other one.
+    [FIRApp configureWithName:appName options:options];
+    FIRApp *newApp = [FIRApp appNamed:appName];
+    FIRFirestore *newInstance = [FIRFirestore firestoreForApp:newApp];
```

Comment:
Please add a check that verifies `[FIRFirestore firestoreForApp:app]` returns the same instance above.

Reply:
I'm not quite sure what you mean - do you mean call `[FIRFirestore firestoreForApp:]` twice and ensure the same instance is returned?

Reply:
Yes that's what I mean (at line ~40). It makes it possible to know that the `XCTAssertNotEqualObjects(newInstance, firestore)` is meaningful here because `XCTAssertEqualObjects(newInstance, firestore)` was true absent this work.